### PR TITLE
feat: meta tag component

### LINKED ISSUE #55: Feature request: `<Meta />` component
**Is your feature request related to a problem? Please describe.**
The blog, including the index and individual posts, has meta tags, which are a strong base for future meta tags support but are manually written.

**Describe the solution you'd like**
A `<Meta />` component that can be reused anywhere on the website.

**Additional context**
Spec:
- [ ] `<Meta />` component with reasonable defaults and properties such as `title`, `description`, etc.
- [ ] Comments sectioning meta tags by platform. For example, see [this](https://www.heymeta.com/url/www.youtube.com/watch?v=iuJDhFRDx9M) and click **Generate**.
- [ ] Add the new component wherever applicable on the website.

### PR DESCRIPTION
**Pull request by niklasmtj**
This fixes #55. 

I added the newly created `Meta` component also to the `Head` component which is already used on a couple of pages. 

Happy to about any feedback about the component.

### CODE REVIEW

File: components/Meta.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+    <>
+      {/* HTML Meta Tags */}
+      <title>{title || SITE_NAME}</title>
+      <meta
+        name="description"
+        content={description || SITE_DESCRIPTION}
+      />
+      <meta property="og:type" content="website" />
+      <meta property="og:site_name" content={title || SITE_NAME} />
+      <meta property="og:locale" content="en" />
```
Comment (iuioiua) [inline components/Meta.tsx:24]: I believe these 3 lines should be moved down to the Facebook Meta Tags section, and `og:site_name` should just have a value of `SITE_NAME`.

File: components/Meta.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+        content={title || SITE_NAME}
+      />
+      <meta
+        property="og:description"
+        content={description || SITE_DESCRIPTION}
+      />
+      <meta
+        property="og:image"
+        content={imageUrl}
+      />
```
Comment (iuioiua) [inline components/Meta.tsx:57]: `imageUrl` is optional. I think it's best to only generate this and other `imageUrl`-containing elements, with a conditional that checks for the existence of `imageUrl` first. E.g. `{imageUrl && <Element>}`.

This behaviour should also occur for other properties of `MetaProps` that lack a default value.

File: components/Meta.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
+// Copyright 2023 the Deno authors. All rights reserved. MIT license.
+import { SITE_DESCRIPTION, SITE_NAME } from "@/constants.ts";
+
+interface MetaProps {
+  title?: string;
+  description?: string;
+  imageUrl?: string;
+  url?: string;
```
Comment (iuioiua) [inline components/Meta.tsx:8]: Are you able to change this to `href`? It might be confusing since the value passed to this parameter is `URL.href`.

**Review by iuioiua — CHANGES_REQUESTED** [review]
Looking good, except for a couple of things. Can you also add somewhere in the README that these meta tags are a feature? A single line should do.

Note: we'll have to validate these meta tags once this PR is pushed to the public website.

File: components/Meta.tsx (hunk 2 above)
Comment (niklasmtj) [reply components/Meta.tsx]: That's what I also though about. Did not know how Twitter etc. reacts when there is a tag without a value. Definitely makes sense to render these conditionally.

**Review by iuioiua — APPROVED** [review]
I love this! Thank you, @niklasmtj.

**Pull request merged**